pkg/conversion/generator.go

```diff
@@ -340,10 +364,62 @@ func (g *generator) writeConversionForSlice(w io.Writer, inField, outField refle
 	return nil
 }
 
-func (g *generator) writeConversionForStruct(w io.Writer, inType, outType reflect.Type, indent int) error {
-	if err := writeHeader(w, g.typeName(inType), g.typeName(outType), indent); err != nil {
+func (g *generator) writeConversionForPtr(w io.Writer, inField, outField reflect.StructField, indent int) error {
+	switch inField.Type.Elem().Kind() {
+	case reflect.Map, reflect.Ptr, reflect.Slice, reflect.Interface, reflect.Struct:
+		// Don't copy these via assignment/conversion!
+	default:
+		// This should handle pointers to all simple types.
+		assignable := inField.Type.Elem().AssignableTo(outField.Type.Elem())
+		convertible := inField.Type.Elem().ConvertibleTo(outField.Type.Elem())
+		if assignable || convertible {
+			ifFormat := "if in.%s != nil {\n"
+			ifStmt := fmt.Sprintf(ifFormat, inField.Name)
+			if err := writeLine(w, indent, ifStmt); err != nil {
+				return err
+			}
+			newFormat := "out.%s = new(%s)\n"
+			newStmt := fmt.Sprintf(newFormat, outField.Name, g.typeName(outField.Type.Elem()))
+			if err := writeLine(w, indent+1, newStmt); err != nil {
+				return err
+			}
+		}
+		if assignable {
+			assignFormat := "*out.%s = *in.%s\n"
+			assignStmt := fmt.Sprintf(assignFormat, outField.Name, inField.Name)
+			if err := writeLine(w, indent+1, assignStmt); err != nil {
+				return err
+			}
+		} else if convertible {
+			assignFormat := "*out.%s = %s(*in.%s)\n"
+			assignStmt := fmt.Sprintf(assignFormat, outField.Name, g.typeName(outField.Type.Elem()), inField.Name)
+			if err := writeLine(w, indent+1, assignStmt); err != nil {
+				return err
+			}
+		}
+		if assignable || convertible {
+			if err := writeLine(w, indent, "}\n"); err != nil {
+				return err
+			}
+			return nil
+		}
+	}
+
+	assignFormat := "if err := s.Convert(&in.%s, &out.%s, 0); err != nil {\n"
+	assignStmt := fmt.Sprintf(assignFormat, inField.Name, outField.Name)
+	if err := writeLine(w, indent, assignStmt); err != nil {
 		return err
 	}
+	if err := writeLine(w, indent+1, "return err\n"); err != nil {
```

Comment:
Check out http://golang.org/pkg/go/format/#Source for auto-formatting of this code. Then you could ignore tabbing.